Apps/Sandcastle/gallery/Drawing on Terrain.html

```diff
@@ -0,0 +1,185 @@
+    <script type="text/javascript">
+    require.config({
+        baseUrl : '../../../Source',
+        waitSeconds : 60
+    });
+    </script>
+</head>
+<body class="sandcastle-loading" data-sandcastle-bucket="bucket-requirejs.html">
+<style>
+    @import url(../templates/bucket.css);
+
+    #viewChanged, #cameraChanged {
+        display: none;
+        background-color: red;
+        color: white;
+    }
+</style>
+<div id="cesiumContainer" class="fullSize"></div>
+<div id="loadingOverlay"><h1>Loading...</h1></div>
+<div id="toolbar"></div>
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    'use strict';
+//Sandcastle_Begin
+var viewer = new Cesium.Viewer('cesiumContainer',  {
+    selectionIndicator : false,
+    terrainProvider : Cesium.createWorldTerrain()
+});
+// Create a function to place a point given a world position.
+function createPoint(worldPosition) {
+    var point = new Cesium.Entity({
+        position : worldPosition,
+        point : {
+            color : Cesium.Color.WHITE,
+            pixelSize : 5,
+            heightReference: Cesium.HeightReference.CLAMP_TO_GROUND
+        }
+    });
+    viewer.entities.add(point);
+    return point;
+}
+// On left mouse click, we want to:
+// * Start a new shape if one doesn't already exist.
+// * Add a vertex to the active shape if one exists.
+var activeShapePoints = [];
+var activeShape;
+var floatingPoint;
+var drawingMode = 'line';
+var handler = new Cesium.ScreenSpaceEventHandler(viewer.canvas);
+handler.setInputAction(function(event) {
+    // We use `viewer.scene.pickPosition` here instead of `viewer.camera.pickEllipsoid` so that
+    // we get the correct point when mousing over terrain.
+    var earthPosition = viewer.scene.pickPosition(event.position);
+    // `earthPosition` will be undefined if our mouse is not over the globe.
+    if (Cesium.defined(earthPosition)) {
+        // If this is the first point, we need to initialize the shape.
+        if (activeShapePoints.length === 0) {
+            // This is the point that'll follow the mouse.
+            floatingPoint = createPoint(earthPosition);
+            // For the first point, we want to push an extra position into our list of vertices
+            // this will be a dynamic point that'll be replaced when the mouse moves.
+            activeShapePoints.push(earthPosition);
+            // We create a callback property for our positions since we want the shape to update
+            // in real time as we move the mouse. We make sure to pass `false` to the
+            // `CallbackProperty` constructor to mark it as not constant.
+            var dynamicPositions = new Cesium.CallbackProperty(function () {
+                // This is a work-around since polylines when `clampToGround` is true will
+                // crash if you try to draw a zero length line
+                if (activeShapePoints.length === 2) {
```

Comment:
Is this fixed by the latest polyline on terrain PR? If not, can we open an issue if there isn't already one?

Reply:
Just tested it with the fix in https://github.com/AnalyticalGraphicsInc/cesium/pull/6693 and it looks like we don't need this workaround anymore! I'll remove it.